cameo/core/solution.py

```diff
@@ -160,11 +158,17 @@ def __init__(self, model, *args, **kwargs):
         self.fluxes = OrderedDict()
         self.shadow_prices = OrderedDict()
         self.reduced_costs = OrderedDict()
+        self._primal_values = model.solver.primal_values
+        self._dual_values = model.solver.dual_values
```

Comment:
This should be `model.solver.shadow_prices`. The function `dual_values` gives the constraint primal values (which doesn't really make sense and will be changed soon)